src/can.rs

```diff
@@ -0,0 +1,498 @@
+pub use embedded_hal_can::{self, Filter, Frame, Id, Receiver, Transmitter};
+
+use crate::gpio::gpioa;
+use crate::gpio::AF9;
+use crate::rcc::APB1;
+use crate::stm32;
+use arrayvec::ArrayVec;
+use nb;
+use nb::Error;
+
+use core::sync::atomic::{AtomicU8, Ordering};
+
+const EXID_MASK: u32 = 0b11111111111000000000000000000;
+const MAX_EXTENDED_ID: u32 = 0x1FFFFFFF;
```

Comment:
Would be more readable this way:

```suggestion
const EXID_MASK: u32 = 0b11111_11111100_00000000_00000000;
const MAX_EXTENDED_ID: u32 = 0x1FFF_FFFF;
```